src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -3339,8 +3315,16 @@ private IEnumerable<SourceRepository> GetEffectiveSources(IEnumerable<SourceRepo
         {
             // Always have to add the packages folder as the primary repository so that
             // dependency info for an installed package that is unlisted from the server is still available :(
-            var effectiveSources = new List<SourceRepository>(primarySources);
-            effectiveSources.Add(PackagesFolderSourceRepository);
+            var count = (primarySources?.Count() ?? 0) +
+                (secondarySources?.Count() ?? 0)
+                + 1;
+
+            var effectiveSources = new List<SourceRepository>(new SourceRepository[count])
```

Comment:
you aren't assigning capacity to the list named effectiveSources

Reply:
:|